### PR TITLE
Remove useless try/catch in the CachedTrait

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

Sorry, I missed the #1908 review.

### CODE REVIEW

**Comment by soyuka** [comment]
thanks @meyerbaptiste

File: src/Cache/CachedTrait.php [hunk 1]
```diff
@@ -33,26 +33,18 @@ private function getCached(string $cacheKey, callable $getValue)
...
-        if (!isset($cacheItem)) {
-            return $this->localCache[$cacheKey] = $value;
+        if ($cacheItem->isHit()) {
+            return $this->localCache[$cacheKey] = $cacheItem->get();
         }
 
-        try {
-            $cacheItem->set($value);
-            $this->cacheItemPool->save($cacheItem);
-        } catch (CacheException $e) {
```
Comment (bendavies) [inline src/Cache/CachedTrait.php]: I was only copying exciting code in the previous pr so I am not sure of the original author's thoughts, but you are now keeping the exception handling above but removing this one. Should we be consistent?

Comment (teohhanhui) [reply src/Cache/CachedTrait.php]: There is no chance of `CacheException` being thrown here. The original author (i.e. myself) was mistaken. :laughing:

Comment (soyuka) [reply src/Cache/CachedTrait.php]: And we all copied without thinking twice :laughing:

**Pull request merged**